test/testdata/rewriter/initializer.rb

```diff
@@ -0,0 +1,47 @@
+# typed: true
+class Foo
+  extend T::Sig
+
+  sig { params(x: Integer).void }
+  def initialize(x)
+    @x = x  # this will get a synthetic `T.let`
+    @y = x + 1  # This should not, because the RHS is not a simple local
```

Comment:
Some suggestions for tests:

- ivar writes that are not immediately at the top of the method, i.e., after some method calls or something
- ivar writes that are inside `if` blocks
- ivars written out of order w.r.t. their declaration order in the MethodDef
- ivars written to inside a block (like `[].each {}`)
- ivars with more complex types (something nilable, something generic)

Reply:
- class var writes, like `@@x` 
- ivar writes for the singleton class, i.e., in the <static-init> not in initialize

Reply:
I've added all of these. It's important to note that not all of them currently work, but I've written some comments in the test about whether we expect them to or not.